File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1 :
            return 0
        min_buy = prices[0]
        max_profit = 0
        for i in range(len(prices)):
            min_buy = min(min_buy, prices[i])
            max_profit = max(max_profit , prices[i] - min_buy)
        return max_profit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    if (prices.length <= 1) return 0;
    let min = prices[0]
    let max = 0
    for (let i = 0; i < prices.length; i++) {
        min = Math.min(min, prices[i])
        max = Math.max(max, (prices[i] - min))
    }
    return max
};